src/seedu/addressbook/Main.java

```diff
@@ -106,12 +106,15 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
         try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        }catch (StorageFile.StorageReadOnlyException sroe){
```

Comment:
Minor change: do keep in mind the standard for spacing between code and keep it consistent.